controllers/btpoperator_controller.go

```diff
@@ -1131,3 +1201,480 @@ func (r *BtpOperatorReconciler) watchConfigPredicates() predicate.Funcs {
+	if ca == nil || caPk == nil {
+		data, err := r.getDataFromSecret(ctx, CaSecret)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+
+		ca, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix), data)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+
+		caPk, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, RsaKeyPostfix), data)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+	}
+
+	webhookCertificate, webhookPrivateKey, err := certs.GenerateSignedCertificate(expiration, ca, caPk)
+	if err != nil {
+		return []byte{}, nil, err
+	}
+
+	return webhookCertificate, webhookPrivateKey, err
+}
+
+func (r *BtpOperatorReconciler) appendCertificationDataToUnstructured(us *[]*unstructured.Unstructured, certName string, certificate, privateKey []byte, prefix string) error {
+	err, data := r.mapCertToSecretData(certificate, privateKey, r.buildKeyNameWithExtension(prefix, CertificatePostfix), r.buildKeyNameWithExtension(prefix, RsaKeyPostfix))
+	if err != nil {
+		return err
+	}
+	secret := r.buildSecretWithDataAndLabels(certName, data, map[string]string{managedByLabelKey: operatorName})
+
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(secret)
+	if err != nil {
+		return err
+	}
+	*us = append(*us, &unstructured.Unstructured{Object: unstructuredObj})
+	return nil
+}
+
+func (r *BtpOperatorReconciler) mapCertToSecretData(certificate, privateKey []byte, keyNameForCert, keyNameForPrivateKey string) (error, map[string][]byte) {
+	return nil, map[string][]byte{
+		keyNameForCert:       certificate,
+		keyNameForPrivateKey: privateKey,
+	}
+}
+
+func (r *BtpOperatorReconciler) reconcileWebhooks(ctx context.Context, us *[]*unstructured.Unstructured, expectedCa []byte) error {
+	logger := log.FromContext(ctx)
+	logger.Info("starting reconcilation of webhooks")
+
+	if expectedCa == nil {
+		secret := &corev1.Secret{}
+		if err := r.Get(ctx, client.ObjectKey{Namespace: ChartNamespace, Name: CaSecret}, secret); err != nil {
+			return err
+		}
+		ca, ok := secret.Data[r.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix)]
+		if !ok || ca == nil {
+			return fmt.Errorf("while receiving certificate data from ca secret in reconcilation webhook")
+		}
+		expectedCa = ca
+	}
+
+	for i, u := range *us {
+		switch kind := u.GetKind(); kind {
+		case MutatingWebhookConfiguration:
+			{
+				err := r.reconcileWebhook(ctx, &u, expectedCa, MutatingWebhookConfiguration)
+				if err != nil {
+					return err
+				}
+				(*us)[i] = u
+			}
+		case ValidatingWebhookConfiguration:
+			{
+				err := r.reconcileWebhook(ctx, &u, expectedCa, ValidatingWebhookConfiguration)
+				if err != nil {
+					return err
+				}
+				(*us)[i] = u
+			}
+		default:
+		}
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) reconcileCaBundle(u **unstructured.Unstructured, expectedCaBundle []byte) error {
+	if u == nil {
+		return fmt.Errorf("empty u param to reconcileCaBundle")
```

Comment:
1. `u` is cryptic, refers to type, not meaning
2. message refers to name of function parameters